### DISCUSSION TITLE
Use of fuzzy hashing to generate hashes

### DISCUSSION THREAD

**Discussion by PhysicsIsAwesome** [abrahamjuliot/creepjs#160] [Ideas]
Normal hashes have a property called cryptographic diffusion, which leads to large changes in the hash even for small changes in the original entity. For browser fingerprinting something like fuzzy hashing seems more appropriate since similar inputs (fingerprinting properties) lead to similar hashes. With this method a visitor could check the similarity of different browser configurations and is closer to real world applicability of tracking by browser fingerprinting which needs to be robust to small changes.

Does CreepJS use some form of fuzzy hashing to generate hashes?

**Comment by abrahamjuliot** [comment]
I have not looked into fuzzy hashing. That is interesting. Are there any code samples or examples? I could see that being useful.

For fingerprint tracing, I'm using `crypto.subtle.digest('SHA-256', data)`. My goal there is to use a web standard API and avoid collision.

Separately, in the template and in a few controlled metrics, I use a light weight hashing function to show sub section diffs.

**Reply by PhysicsIsAwesome** [reply]
> I have not looked into fuzzy hashing. That is interesting. Are there any code samples or examples?

FingerprintJS uses fuzzy matching in their [Pro](https://fingerprintjs.com/github/) version to link fingerprints and fuzzy hashing is being used to detect [similar files and similar malware](https://www.microsoft.com/security/blog/2021/07/27/combing-through-the-fuzz-using-fuzzy-hashing-and-deep-learning-to-counter-malware-detection-evasion-techniques/), but I don't have further code examples.

**Comment by abrahamjuliot** [comment]
Nice. I'm researching this and digesting this js library.

https://github.com/huwenshuo/ctph.js

**Comment by abrahamjuliot** [comment]
I have a form of fuzzy hashing in the works and it should appear in a feature release.
